### PR TITLE
fix id vector for DAEs

### PR DESCRIPTION
**Pull request by FFroehlich**
Following a post on the SUNDIALS mailing list, I realized that the ID vector is incorrectly implemented. The previous implementation computed the differential *equations* (which was also done incorrectly) and not the *components*.

### CODE REVIEW

File: matlab/@amimodel/parseModel.m [hunk 1]
```diff
@@ -184,7 +184,7 @@ function parseModel(this)
 if(strcmp(this.wtype,'iw'))
     if(isfield(this.fun,'M'))
         this.getFun([],'M');
-        this.id = double(logical(sum(this.fun.M.sym,2)~=0));
+        this.id = double(logical(any(this.fun.M.sym)));
```
Comment (paulstapor) [inline matlab/@amimodel/parseModel.m:187]: Shouldn't it be `any(this.fun.M.sym, 2)` ?

Comment (dweindl) [reply matlab/@amimodel/parseModel.m:187]: That would yield the same result, not? (With `any` potentially being a bit faster - not that it really matters here)

**Review by dweindl — COMMENTED** [review]
As this went unnoticed for quite a while, it might be good to have it tested somewhere. Although I'd rather not add much new matlab code.

File: matlab/@amimodel/parseModel.m (hunk 1 above)
Comment (FFroehlich) [reply matlab/@amimodel/parseModel.m:187]: From my understanding, checking for differential *components* can be done by checking whether any of the respective columns of the mass matrix are nonzero, to see whether the time derivative of the respective component appears *anywhere* in the rhs.

Comment (paulstapor) [reply matlab/@amimodel/parseModel.m:187]: Ah, okay. I see my error. Indeed, your code is fine!

**Review by paulstapor — APPROVED** [review]

File: matlab/@amimodel/parseModel.m [hunk 2]
```diff
@@ -184,7 +184,7 @@ function parseModel(this)
 if(strcmp(this.wtype,'iw'))
     if(isfield(this.fun,'M'))
         this.getFun([],'M');
```
Comment (dweindl) [inline matlab/@amimodel/parseModel.m:186]:
```suggestion
if(strcmp(this.wtype, 'iw'))
    if(isfield(this.fun, 'M'))
        this.getFun([], 'M');
```

Just to trigger GHA...

**Comment by FFroehlich** [comment]
existing examples work, #464 remains unfixed by this.

**Pull request merged**